include/alpaka/block/sync/BlockSyncBarrierOacc.hpp

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2020 Jeffrey Kelling
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#if _OPENACC < 201306
+    #error If ALPAKA_ACC_ANY_BT_OACC_ENABLED is set, the compiler has to support OpenACC xx or higher!
+#endif
+
+#include <alpaka/block/sync/Traits.hpp>
+
+namespace alpaka
+{
+    //#############################################################################
+    //! The OpenACC barrier block synchronization.
+    class BlockSyncBarrierOacc
+    {
+    public:
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST BlockSyncBarrierOacc() = default;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST BlockSyncBarrierOacc(BlockSyncBarrierOacc const &) = delete;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST BlockSyncBarrierOacc(BlockSyncBarrierOacc &&) = delete;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator=(BlockSyncBarrierOacc const &) -> BlockSyncBarrierOacc & = delete;
+        //-----------------------------------------------------------------------------
+        ALPAKA_FN_HOST auto operator=(BlockSyncBarrierOacc &&) -> BlockSyncBarrierOacc & = delete;
+        //-----------------------------------------------------------------------------
+        ~BlockSyncBarrierOacc() = default;
+
+        std::uint8_t mutable m_generation = 0u;
+        // NVHPC 20.7: initializer causes warning:
+        // NVC++-W-0155-External and Static variables are not supported in acc routine - _T139951818207704_37530
+        int mutable m_syncCounter[4] {0,0,0,0};
```

Comment:
Where does the number 4 come from?

Reply:
Space for two generation times two slots in https://github.com/jkelling/alpaka/blob/openacc/include/alpaka/ctx/block/CtxBlockOacc.hpp#L105 .

Reply:
Thx! Could you please add a comment for that or move 4 into a named constant?

Reply:
A constant would suggest safety which is not there, also: I cannot come up with a descriptive name. So, a comment it is.